tests/testutil/testutil.go

```diff
@@ -213,3 +217,57 @@ func homeDir() string {
 	}
 	return os.Getenv("USERPROFILE") // windows
 }
+
+// CreateCNAME creates a CNAME if it doesn't exist or update if it exist already in the DNS Zone
+// We use AWS Route53 as DNS provider.
+func CreateCNAME(clusterDomain string) *route53.ChangeResourceRecordSetsOutput {
+	sess, err := session.NewSession()
+	if err != nil {
+		fmt.Println("failed to create session make sure to add credentials to ~/.aws/credentials,", err)
+		fmt.Println(err.Error())
+	}
+	cname := GetCNAMEInfo()
+	zoneID := getZoneIDInfo()
+	if cname == "" || clusterDomain == "" || zoneID == "" {
+		fmt.Println(fmt.Errorf("incomplete information: d: %s, t: %s, z: %s", cname, clusterDomain, zoneID))
+	}
+	svc := route53.New(sess)
+
+	params := &route53.ChangeResourceRecordSetsInput{
+		ChangeBatch: &route53.ChangeBatch{
+			Changes: []*route53.Change{
+				{
+					Action: aws.String("UPSERT"),
+					ResourceRecordSet: &route53.ResourceRecordSet{
+						Name: aws.String(cname),
+						Type: aws.String("CNAME"),
+						ResourceRecords: []*route53.ResourceRecord{
+							{
+								Value: aws.String(clusterDomain),
+							},
+						},
+						TTL: aws.Int64(10),
+					},
+				},
+			},
+			Comment: aws.String("Add CNAME to Rio Cluster Domain"),
+		},
+		HostedZoneId: aws.String(zoneID),
+	}
+	resp, err := svc.ChangeResourceRecordSets(params)
```

Comment:
Does this get cleaned up in AWS after test runs ?

Reply:
it's an upsert if the CNAME domain exist it will update it if not this will create it. I can do the clear, but for testing locally this worked fine.

Reply:
Need to make sure we aren't leaving a bunch of test records around.

Reply:
I've added Delete CNAME, and rebased with master.